test/GasAccounting.t.sol

```diff
@@ -132,9 +144,62 @@ contract GasAccountingTest is Test {
         assertEq(executionEnvironment.balance, borrowedAmount);
     }
 
+    function test_borrow_phasesEnforced() public {
+        // borrow should revert if called in or after PostSolver phase
+
+        uint256 borrowedAmount = 1e18;
+        deal(address(mockGasAccounting), borrowedAmount);
+        assertEq(executionEnvironment.balance, 0, "EE should start with 0 ETH");
+        uint256 startState = vm.snapshot();
+
+        // Allowed borrowed phases: PreOps, UserOperation, PreSolver, SolverOperations
+
+        mockGasAccounting.setPhase(ExecutionPhase.PreOps);
+        vm.prank(executionEnvironment);
+        mockGasAccounting.borrow(borrowedAmount);
+        assertEq(executionEnvironment.balance, borrowedAmount);
+        vm.revertTo(startState);
+
+        mockGasAccounting.setPhase(ExecutionPhase.UserOperation);
+        vm.prank(executionEnvironment);
+        mockGasAccounting.borrow(borrowedAmount);
+        assertEq(executionEnvironment.balance, borrowedAmount);
+        vm.revertTo(startState);
+
+        mockGasAccounting.setPhase(ExecutionPhase.PreSolver);
+        vm.prank(executionEnvironment);
+        mockGasAccounting.borrow(borrowedAmount);
+        assertEq(executionEnvironment.balance, borrowedAmount);
+        vm.revertTo(startState);
+
+        mockGasAccounting.setPhase(ExecutionPhase.SolverOperations);
+        vm.prank(executionEnvironment);
+        mockGasAccounting.borrow(borrowedAmount);
+        assertEq(executionEnvironment.balance, borrowedAmount);
+        vm.revertTo(startState);
+
+        // Disallowed borrowed phases: PostSolver, AllocateValue, PostOps
+
+        mockGasAccounting.setPhase(ExecutionPhase.PostSolver);
+        vm.prank(executionEnvironment);
+        vm.expectRevert(AtlasErrors.WrongPhase.selector);
+        mockGasAccounting.borrow(borrowedAmount);
+        vm.revertTo(startState);
+
+        mockGasAccounting.setPhase(ExecutionPhase.AllocateValue);
+        vm.prank(executionEnvironment);
+        vm.expectRevert(AtlasErrors.WrongPhase.selector);
+        mockGasAccounting.borrow(borrowedAmount);
+        vm.revertTo(startState);
+
+        mockGasAccounting.setPhase(ExecutionPhase.PostOps);
+        vm.prank(executionEnvironment);
+        vm.expectRevert(AtlasErrors.WrongPhase.selector);
+        mockGasAccounting.borrow(borrowedAmount);
+        vm.revertTo(startState);
+    }
```

Comment:
shouldn't there be some sort of assertion here?

Reply:
Added assert balance is 0 because borrow should have reverted https://github.com/FastLane-Labs/atlas/pull/227/commits/505b153331dacef31a2ce9324e90865397c39902